base/system/smss/pagefile.c

```diff
@@ -20,8 +20,59 @@
 //
 #define STANDARD_PAGING_FILE_NAME       L"\\??\\?:\\pagefile.sys"
 #define STANDARD_DRIVE_LETTER_OFFSET    4
-#define MEGABYTE                        0x100000UL
-#define MAXIMUM_PAGEFILE_SIZE           (4095 * MEGABYTE)
+#define MAX_PAGING_FILES                16  // See also ntoskrnl/include/internal/mm.h
+#define MEGABYTE                        (1024 * 1024)
+
+/* Minimum pagefile size is 256 pages (1 MB) */
+// #define MINIMUM_PAGEFILE_SIZE           (256ULL * PAGE_SIZE)
+
+/* Maximum pagefile sizes for different architectures */
+#define GIGABYTE                        (1024ULL * MEGABYTE)
+#define TERABYTE                        (1024ULL * GIGABYTE)
+
+// NOTE: No changes for NTDDI_WIN10
+#if (NTDDI_VERSION >= NTDDI_WINBLUE) // NTDDI_WIN81
+#define MAXIMUM_PAGEFILE_SIZE32         ((1ULL * 1024 * 1024 - 1) * PAGE_SIZE)
+                                     // PAGE_ROUND_DOWN(4ULL * GIGABYTE - 1)
+#else
+/* 4095 MB */
+#define MAXIMUM_PAGEFILE_SIZE32         (4095ULL * MEGABYTE)
+#endif
+
+// NOTE: No changes for NTDDI_WIN10
+#if (NTDDI_VERSION >= NTDDI_WINBLUE) // NTDDI_WIN81
+#define MAXIMUM_PAGEFILE_SIZE64         ((4ULL * 1024 * 1024 * 1024 - 1) * PAGE_SIZE)
+                                     // PAGE_ROUND_DOWN(16ULL * TERABYTE - 1)
+#else
+/* 16 TB */
+#define MAXIMUM_PAGEFILE_SIZE64         (16ULL * TERABYTE)
+#endif
+
+#if defined(_M_IX86)
+    #define MAXIMUM_PAGEFILE_SIZE       MAXIMUM_PAGEFILE_SIZE32
+    /* PAE uses the same size as x64 */
+    #define MAXIMUM_PAGEFILE_SIZE_PAE   MAXIMUM_PAGEFILE_SIZE64
+#elif defined (_M_AMD64) || defined(_M_ARM64)
+    #define MAXIMUM_PAGEFILE_SIZE       MAXIMUM_PAGEFILE_SIZE64
+#elif defined (_M_IA64)
+/* 32 TB */
+    #define MAXIMUM_PAGEFILE_SIZE       (32ULL * TERABYTE)
+#elif defined(_M_ARM)
+/* Around 2 GB */
+    // NOTE: No changes for NTDDI_WIN10
+    #if (NTDDI_VERSION >= NTDDI_WINBLUE) // NTDDI_WIN81
+    #define MAXIMUM_PAGEFILE_SIZE       ((512ULL * 1024 - 1) * PAGE_SIZE)
+                                     // PAGE_ROUND_DOWN(2ULL * GIGABYTE - 1)
+    #else
+/* 4095 MB */
+    #define MAXIMUM_PAGEFILE_SIZE       (4095ULL * MEGABYTE)
+    #endif
+#else
+/* On unknown architectures, default to 4095 MB */
+#pragma message("Unknown architecture")
+    #define MAXIMUM_PAGEFILE_SIZE       (4095ULL * MEGABYTE)
+#endif
```

Comment:
Maybe do here `#if defined(_WIN64) ... #else ...`

Reply:
Just for the ARM case?

Reply:
No I mean this one for the unknown case.